backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/workpackage/impl/AbstractContractLog.java

```diff
@@ -165,6 +172,25 @@ public AbstractContractLog(@NonNull final ModularContractService modularContract
 	@Override
 	public @NonNull ExplainedOptional<LogEntryReverseRequest> createLogEntryReverseRequest(@NonNull final CreateLogRequest createLogRequest)
 	{
-		throw new AdempiereException(MSG_ERROR_DOC_ACTION_UNSUPPORTED);
+		final TableRecordReference recordRef = createLogRequest.getRecordRef();
+		final I_C_Flatrate_Term contract = flatrateBL.getById(FlatrateTermId.ofRepoId(recordRef.getRecordIdAssumingTableName(getSupportedTableName())));
```

Comment:
This one looks very awkward.
i think you should
* assume getSupportedTableName() is flat rate term
* call org.adempiere.util.lang.impl.TableRecordReference#getIdAssumingTableName

Reply:
It's a private field in this class, so I could change it too, if that helps
```java
recordRef.getRecordIdAssumingTableName(supportedTableName)
```
Also the same field is used to check if this handler is applicable etc. (`de.metas.contracts.modular.workpackage.ModularContractLogHandlerRegistry#applies`) So if someone changes this one it compelitly breakes. If we check it here, we also should adjust it in each of the other log handlers.